src/core/qgstaskmanager.cpp

```diff
@@ -41,6 +43,7 @@ QgsTask::~QgsTask()
     delete subTask.task;
   }
   mNotFinishedMutex.unlock();
+  mNotStartedMutex.release();
```

Comment:
The qt docs aren't clear -- what happens if there's no resources acquired when this is called?

Reply:
I tested it on Linux and Windows. if there's no resources acquired when release is called, the resources available is increased (by the number given in release call, 1 in this case) and if there is threads waiting on acquire, they wake up.

In this case, there is no thread to wake up because task has to be started or canceled before being deleted by the task manager. And both way release the semaphore and wake up a possible waitForFinished call.

Reply:
Ok, let's give it a shot!!